### PR TITLE
Use && for splitting long texts to multiple lines and aliases for interface members

### PR DESCRIPTION
**Pull request by shafiq2601**
Updated as per comments in the previous pull request.

### CODE REVIEW

File: clean-abap/CleanABAP.md [hunk 1]
```diff
@@ -2181,6 +2200,30 @@ which will never have an alternative implementation and will never be mocked in
...
+```ABAP
+class clean_class definition ...
+public section.
+  interfaces legacy_interface_long_name.
+private section.
+  aliases result for legacy_interface_long_name~result .
+  aliases input  for legacy_interface_long_name~input  .
+  aliases get_something for legacy_interface_long_name~get_something .
+ ....
+ ```
```
Comment (schmelto) [inline clean-abap/CleanABAP.md:2218]:
```suggestion
```
```

File: clean-abap/CleanABAP.md [hunk 2]
```diff
@@ -2181,6 +2200,29 @@ which will never have an alternative implementation and will never be mocked in
...
+
+```ABAP
+class clean_class definition ...
+public section.
+  interfaces legacy_interface_long_name.
+private section.
+  aliases result for legacy_interface_long_name~result .
+  aliases input  for legacy_interface_long_name~input  .
+  aliases get_something for legacy_interface_long_name~get_something .
+ ....
```
Comment (schmelto) [inline clean-abap/CleanABAP.md:2217]:
```diff
+```
```

sorry I've made an mistake in my review comment. I just wanted to show that there was a space which was not needed and not that the whole line is not mandatory 😄

**Comment by lucasborin** [comment]
Personally, I am not a fan of the `ALIASES`. I usually recommend the following approach:

For instance:
image

The class returns its instance typed as the interface so that the caller doesn't need to call the `interface~method`.

Ps: singleton is optional.

**Comment by jordao76** [comment]
There are two concepts in this pull request. Ideally, create two pull requests so that they can be reviewed individually.

As for the aliases, like mentioned by @lucasborin, you should use a reference to the interface type instead of the class if possible.

**Comment by pokrakam** [comment]
I agree with concept 1 to split strings, but I am not in favour of the aliases part (separate PRs are better). 

Long names may be inconvenient at times, but for attributes they serve a reminder that you are assigning public API attributes. For methods, a class calling its own API can be a code smell so classes should rarely be calling methods of interfaces they implement. 

Aliases have their own problems and make refactoring fragile, see discussion at #176 

Personally I rarely see this as a problem, so in my opinion people can do it in private and where it makes sense, but am not in favour of putting it in the style guide as that would suggest creating them by default.

**Review by N2oB6n-SAP — CHANGES_REQUESTED** [review]
**:warning: Please split the PR as it presents two separate concerns (interface aliasing recommendation and &&-splitting).**

---

**About the &&-splitting:**
This might be added to the _Strings_ section to illustrate how to apply _Stick to a reasonable line length_ to string assembly. It can serve as a valuable reminder how to adhere to a line length limit and thus complements the preceding rules. However, I would not go beyond and advise for "short" and "crisp" texts as this is a matter of content and not of code structure/readability. Long texts/templates should ideally be placed elsewhere (i18n, ...), anyways, and not be part of the source.
To generalize I would favor adding a new section titled "_Use & and && to split texts over multiple lines_" since depending on what you're dealing with `&` or `&&` should be chosen. (Note that string templates work with both and for simple string expressions `&` may be optimized at compile-time.)

**About the aliasing interfaces:**
I disagree with this proposal. Using aliases complicates working with the where-used list and makes it harder to refactor on a greater scale. Also, when properly typing object-referencing variables/members they can easily be hidden in a local context. Having to remap an interface alias to its full name could be considered unecessary mental overhead with little benefit in return, except for the savings after the `METHOD` (implementation bracket).

**Comment by HrFlorianHoffmann** [comment]
We'll try to split this PR.

**Comment by sautermi0** [comment]
FYI - I think this is the previous PR: https://github.com/SAP/styleguides/pull/212

**Comment by HrFlorianHoffmann** [comment]
@shafiq2601 Hi Shafiquel! Tried to contact you but couldn't find up-to-date contact information, as you seem to have left SAP Labs in the meantime.

If you are still interested in this topic, could you split this Pull Request into two as advised above by N2oB6n-SAP? We'd like to merge the part with the "&&" concatenation, and splitting would be the easiest way to retain your contribution in GitHub's history. The second topic needs more discussion and we'd like to address it in a separate Pull Request.

If I don't hear from you, I'd go ahead and clone the branch to split this on my own. I am not sure I can retain your name in the history by this, then, however.

**Comment by HrFlorianHoffmann** [comment]
- Couldn't reach the original author
- We'd like to split this, keeping the "Use && to split long text part",
- But discarding the alias part as we prefer the way it is without
- Split this PR by myself, in https://github.com/SAP/styleguides/pull/298
- Mentioning Shafiquel as co-author in the commit message, as described in https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors
- Closing this PR

**Pull request closed without merging**